src/realtime/realtime-module.ts

```diff
@@ -0,0 +1,245 @@
+import { Observable, Subscription } from '@reactivex/rxjs';
+import { inject, injectable } from 'inversify';
+import { isNil, omitBy } from 'lodash';
+
+import {
+  createActivityEvent,
+  MinardActivity,
+} from '../activity';
+import {
+  CommentAddedEvent,
+  CommentDeletedEvent,
+  createCommentAddedEvent,
+  createCommentDeletedEvent,
+} from '../comment';
+import {
+  createDeploymentEvent,
+  DeploymentEvent,
+} from '../deployment';
+import { eventBusInjectSymbol, PersistentEventBus } from '../event-bus/';
+import { ApiProject, JsonApiHapiPlugin } from '../json-api';
+import {
+  ApiEntity,
+  toApiBranchId,
+  toApiCommitId,
+  toApiDeploymentId,
+} from '../json-api';
+import {
+  codePushed,
+  CodePushedEvent,
+  projectCreated,
+  ProjectCreatedEvent,
+  projectDeleted,
+  ProjectDeletedEvent,
+  projectEdited,
+  ProjectEditedEvent,
+} from '../project';
+import {
+  Event,
+  eventCreator,
+  isPersistedEvent,
+  isType,
+  PersistedEvent,
+  StreamingEvent,
+} from '../shared/events';
+import * as logger from '../shared/logger';
+import {
+  StreamingCodePushedEvent,
+  StreamingCommentDeletedEvent,
+  StreamingDeploymentEvent,
+} from './types';
+
+@injectable()
+export class RealtimeModule {
+
+  public static injectSymbol = Symbol('realtime-module');
+  private eventBusSubscription: Subscription;
+  private readonly sseStream: Observable<PersistedEvent<any>>;
+
+  constructor(
+    @inject(JsonApiHapiPlugin.injectSymbol) private readonly jsonApiPlugin: JsonApiHapiPlugin,
+    @inject(eventBusInjectSymbol) private readonly eventBus: PersistentEventBus,
+    @inject(logger.loggerInjectSymbol) private readonly logger: logger.Logger,
+  ) {
+
+    // creates SSEEvents and posts them
+    this.eventBusSubscription = this.getEnrichedStream(this.eventBus.getStream())
+      .subscribe(this.eventBus.post.bind(this.eventBus));
+
+    // listens to SSEEvents and shares them
+    this.sseStream = this.eventBus.getStream()
+      .filter(isPersistedEvent)
+      .map(event => event as PersistedEvent<any>)
```

Comment:
Right. I'd like to say the typings for `filter` have improved, making the map unnecessary.